dpcpp/distributed/partition_helpers_kernels.dp.cpp

```diff
@@ -74,11 +74,31 @@ void sort_by_range_start(
         range_start_ends.get_data(), stride{});
     auto end_it = oneapi::dpl::make_permutation_iterator(
         range_start_ends.get_data() + 1, stride{});
+
+    // older versions of oneDPL have a bug when sorting permutation iterators
+#if ONEDPL_VERSION_MAJOR >= 2022 && ONEDPL_VERSION_MINOR >= 1
```

Comment:
I think we should probably move towards a stable version of oneDPL soon, then we could remove this workaround.

Reply:
We are still testing with and older version, so let's keep it.